### DISCUSSION TITLE
Feature request: Isolation of Superset elements (e.g. datasets and charts) through groups/namespaces

### DISCUSSION THREAD

**Discussion by GerbenvdHuizen** [apache/superset#21863] [Ideas]
### Motivation
Currently it is not possible to create Superset elements such as charts and dashboards within a certain namespace/group that only users with access to that namespace can access. For example, if you have multiple teams within an organization, you would want someone who is a member of one of these teams to automatically have access to all the elements related/assigned to their team. To achieve this with the current permission/role setup would be difficult, since there is no proper way to assign a chart or dashboard to a group of users when such an element is created.

### Proposed Change
I propose the introduction of proper group management within Superset through some kind of namespacing/groupings of elements that users can be assigned to. This kind of namespacing would provide a mechanism for isolating groups of Superset elements. Therefore, a user would only be able to access Superset elements that are within a namespace/group they can access. When a Superset element is created by a user it will need to be assigned to their default namespace/group. It would still need to be possible to share elements like it is now, even if you are in a different team.

### Inspiration
Inspired by namespaces in Kubernetes https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/ and group management in gcp https://cloud.google.com/iam/docs/groups-in-cloud-console

**Comment by kamalkeshavani-aiinside** [comment]
Preset has done something like this with workspace feature.
https://docs.preset.io/docs/about-workspaces
But it is their unique(proprietary) implementation, so someone else needs to implement this in Superset code from scratch.
